sirepo/simulation_db.py

```diff
@@ -1176,22 +1296,34 @@ def _validate_number(val, sch_field_info):
             raise AssertionError(util.err(sch_field_info, 'numeric value {} out of range', val))
```

Comment:
The code above should use assert statements:

```py
assert fv <= fmax, \
    util.err(field_info, 'numeric value {} out of range', val)
```

For some reason I can't comment on the code above in GitHub (only on changed lines or near the change?). I thin the if len(field_info) > 5 should be a short circuit exit.

I think the "float" conversions should be modularized so you can say something like:

```
_assert_field_info(val, field_info, 5)
```

I actually don't understand the modularization especially the code:

```py
            sch_field_info = sch_model[field_name]
            _validate_enum(val, sch_field_info, sch_enums)
            _validate_number(val, sch_field_info)
```

It seems to me that something is either an enum or a number, not both.

Reply:
I wasn't clear about assert_field_info. The use of assert_field_info should be simpler in validate_number (which really should be assert_number):

```py
_assert_field_info(field_info, 4, v, float.__ge__, 'less than min')
_assert_field_info(field_info, 5, v, float.__le__, 'great than max')

def _assert_field(field_info, i, v, op, msg):
    try:
        m = float(field_info(i))
    except ValueError:
        return
    assert op(v, m), \
        util.err(v, 'value is {}={}', msg, m)
```

Perhaps this might be better called _assert_bound.